databricks/sdk/_widgets/IPyWidgetUtils.py

```diff
@@ -0,0 +1,89 @@
+import typing
+
+from IPython.core.display_functions import display
+from ipywidgets.widgets import (ValueWidget, Widget, widget_box,
```

Comment:
Users can access dbutils from 2 points
  a. `from databricks.sdk.runtime import *`: This always falls back to the dbr version of dbutils. The oss implementations do not come into effect. 
  b. `workspace_client.dbutils`: This always uses the open source version of dbutils. We are working on a fix so that this property also falls back to the dbr version of dbutils when possible. 
This means that any imports for ipywidgets and other deps do not take effect when on dbr. 

Further, I added a new install group (`databricks-sdk[notebook]`), which has certain pinned versions for widgets and ipython. Users are prompted to install it if we detect that users are in an interactive notebook environment, but by default this will not be installed on the cluster. These checks are also bypassed in the dbr in favour of dbr native dbutils.